buffer/src/test/java/io/netty/buffer/AbstractByteBufTest.java

```diff
@@ -2229,20 +2248,25 @@ public void run() {
         latch.await(10, TimeUnit.SECONDS);
         barrier.await(5, TimeUnit.SECONDS);
         assertNull(cause.get());
+        buffer.release();
     }
 
     @Test(expected = IndexOutOfBoundsException.class)
     public void readByteThrowsIndexOutOfBoundsException() {
-        final ByteBuf buffer = releaseLater(newBuffer(8));
-        buffer.writeByte(0);
-        assertEquals((byte) 0, buffer.readByte());
-        buffer.readByte();
+        final ByteBuf buffer = newBuffer(8);
+        try {
+            buffer.writeByte(0);
+            assertEquals((byte) 0, buffer.readByte());
+            buffer.readByte();
+        } finally {
```

Comment:
@normanmaurer - I don't think we have to retroactively change all tests ... just be consistent in this PR, and as we implement new tests or change existing ones use the same approach. You added a finally block for a few tests in this PR ... can you remove those?